pkgs/build-support/dotnet/add-nuget-deps/default.nix

```diff
@@ -6,12 +6,39 @@
   substituteAll,
   nuget-to-nix,
   nixfmt-rfc-style,
+  nuget-to-json,
   cacert,
   fetchNupkg,
   callPackage,
 }:
 
 {
+  /**
+    A list of nuget packages.
+
+    Should be a list of calls to `fetchNuGet`:
+
+    ```nix
+    { fetchNuGet }:
+    [
+      (fetchNuGet { pname = "AsyncIO"; version = "0.1.69"; hash = "sha256-JQKq/U71NQTfPuUqj7z5bALe+d7G1o3GcI8kvVDxy6o="; })
+    ]
+    ```
+
+    or a JSON file with arguments to `fetchNupkg`:
```

Comment:
Maybe we drop documentation for the legacy pattern here?

If not, it should
1. mention JSON _before_ nix
2. Replace _"Should be a list of calls to"_ with _"Should be a function that returns a list of derivations"_
3. Mention that this pattern is deprecated